examples/openai_embedding_client.py

```diff
@@ -13,11 +13,19 @@
 models = client.models.list()
 model = models.data[0].id
 
-responses = client.embeddings.create(input=[
+input = [
     "Hello my name is",
     "The best thing about vLLM is that it supports many different models"
-],
-                                     model=model)
+]
 
-for data in responses.data:
+responses_float = client.embeddings.create(input=input, model=model, 
+                                           encoding_format="float")
+
+responses_base64 = client.embeddings.create(input=input, model=model, 
+                                            encoding_format="base64")
+
+
+assert responses_float.data == responses_base64.data
```

Comment:
From my understanding, the returned data should be base64 encoded if you pass `encoding_format="base64"`, so you should be decoding its output before comparing them.

Reply:
> From my understanding, the returned data should be base64 encoded if you pass `encoding_format="base64"`, so you should be decoding its output before comparing them.

Interestingly, I manually ran the unit test (with `encoding_format = base64`, e.g. `python3 openai_embedding_client.py`) and added a `print(obj)` at https://github.com/openai/openai-python/blob/main/src/openai/resources/embeddings.py#L99 (on client side), the output (from `print(obj)` ) was actually array of `float`, ***rather than*** array of `base64` string.

The embedding model I use is: https://huggingface.co/intfloat/e5-mistral-7b-instruct

Reply:
Hmm, perhaps OpenAI client automatically performs the decoding for you? Nevertheless, we should still perform the encoding on server-side.

Reply:
> Nevertheless, we should still perform the encoding on server side.

Which part in https://github.com/vllm-project/vllm/blob/main/vllm/entrypoints/openai/serving_embedding.py should we perform the encoding? My understanding is: the input is the array of string https://github.com/vllm-project/vllm/blob/main/vllm/entrypoints/openai/serving_embedding.py#L90

with `encoding_format=base64`, from my test, the response from the server (vllm) looks like already array of `float` https://github.com/vllm-project/vllm/blob/main/vllm/entrypoints/openai/serving_embedding.py#L35 (`print(final_res.outputs.embedding)`).

Thanks for the hints

Reply:
We should only apply base64 encoding on the outputs.

Reply:
Tried to encode the embedding in the response (the latest code checked in), however, adding the base64 encoding failed openai client validation (as following),
```
Traceback (most recent call last):
  File "/Users/xxx/vllm/examples/openai_embedding_client.py", line 34, in <module>
    responses_base64 = client.embeddings.create(input=input, model=model, 
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/bytedance/python-env/lib/python3.11/site-packages/openai/resources/embeddings.py", line 117, in create
    return self._post(
           ^^^^^^^^^^^
  File "/Users/bytedance/python-env/lib/python3.11/site-packages/openai/_base_client.py", line 1250, in post
    return cast(ResponseT, self.request(cast_to, opts, stream=stream, stream_cls=stream_cls))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/bytedance/python-env/lib/python3.11/site-packages/openai/_base_client.py", line 931, in request
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/bytedance/python-env/lib/python3.11/site-packages/openai/_base_client.py", line 1030, in _request
    raise self._make_status_error_from_response(err.response) from None
openai.BadRequestError: Error code: 400 - {'object': 'error', 'message': "1 validation error for EmbeddingResponseData\nembedding\n  Input should be a valid list [type=list_type, input_value=b'AAAAAADchD8AAAAAABCUPwA...wAAAAAA6JC/AAAAAAAIgD8=', input_type=bytes]\n    For further information visit https://errors.pydantic.dev/2.7/v/list_type", 'type': 'BadRequestError', 'param': None, 'code': 400}
```

It looks like the validation only except the `number`, instead of the 'string' in base64?

Reply:
Can you use a debugger to figure out why the code which I linked above is not running to decode the base64 output?

Reply:
> https://github.com/openai/openai-python/blob/main/src/openai/resources/embeddings.py#L102-L110

Yeah - at https://github.com/openai/openai-python/blob/main/src/openai/resources/embeddings.py#L98, as we explicitly give `encoding_format = base64`,  it will return the object without further decoding. However, the returned obj is array of `float`.

(Maybe I am wrong) it seems whatever the encoding_format is, there is no impact on the input to the server (vllm) https://github.com/vllm-project/vllm/blob/main/vllm/entrypoints/openai/serving_embedding.py#L90.  From many tests, look like the model I use (https://huggingface.co/intfloat/e5-mistral-7b-instruct) always output the array of `float`

Reply:
From my understanding,, the base64 encoding should only be applied to the output, not the input.